facilitator/src/lib.rs

```diff
@@ -39,15 +45,57 @@ pub enum Error {
     HttpError(#[from] ureq::Error),
 }
 
-/// An implementation of transport::TransportWriter that computes a SHA256
-/// digest over the content it is provided.
-pub struct DigestWriter {
+pub fn generate_validation_packet(
+    servers: &mut Vec<Server<Field32>>,
+    packet: &IngestionDataSharePacket,
+) -> Result<ValidationPacket> {
+    let r_pit = Field32::from(
+        u32::try_from(packet.r_pit)
+            .with_context(|| format!("illegal r_pit value {}", packet.r_pit))?,
+    );
+
+    // TODO(timg): if this fails for a non-empty subset of the
+    // ingestion packets, do we abort handling of the entire
+    // batch (as implemented currently) or should we record it
+    // as an invalid UUID and emit a validation batch for the
+    // other packets?
+    for server in servers.iter_mut() {
+        let validation_message = match server
+            .generate_verification_message(r_pit, &packet.encrypted_payload)
+        {
+            Ok(m) => m,
+            Err(ServerError::Encrypt(_)) => {
```

Comment:
It's a bit of a shame that we just drop the error here, but this was a problem before your PR. Filed #1033.